ui/StatusQ/src/StatusQ/Components/private/chart/Library/Library.js

```diff
@@ -0,0 +1,135 @@
+.pragma library
+
+.import "Polyfills.js" as Polyfills
+.import "Chart.bundle.js" as ChartJs
+.import "chartjs-plugin-crosshair.js" as CrosshairLib
+.import "chartjs-plugin-datalabels.js" as DataLabelsLib
+
+
+/*! 
+    /This file is used to provide the necessary wrappers for Chart.js to work in QML
+    /It is loaded after Chart.js and plugins and provides the necessary functions
+    /NOTE: Some plugins work out of the box, others need to be adapted to work in QML
+!*/
+
+(function(global){
+    
+    var color = global.Color
+    Chart.helpers.merge(Chart.defaults.global, {
+        // Default options
+        events: [
+            "mousemove",
+            "mouseout",
+            "click"
+        ]
+    })
+
+    // QML rendering plugin
+    Chart.plugins.register({
+        afterDraw: function(chart) {
+            chart.canvas.requestPaint()
+        }
+    })
+
+    var MOUSE_EVENTS = {
+        click: "clicked",
+        mousemove: "positionChanged",
+        mouseenter: "entered",
+        mouseout: "exited",
+        mousedown: "pressed",
+        mouseup: "released"
+    }
+
+    function createEvent(type, chart, x, y, native, target) {
+        return {
+            type: type,
+            chart: chart,
+            native: native || null,
+            x: x !== undefined ? x : null,
+            y: y !== undefined ? y : null,
+            target: target || null,
+        }
+    }
+
+    // QML platform implementation
+    Chart.helpers.merge(Chart.platform, {
+        addEventListener: function(chart, type, listener) {
+            var canvas = chart.canvas
+            if (type === "resize") {
+                canvas.resized.connect(function() {
+                    listener(createEvent(type, chart))
+                })
+                return
+            }
+
+            var mapped = MOUSE_EVENTS[type]
+            if (!mapped) {
+                console.warn("Unsupported event:", type)
+                return
+            }
+
+            canvas._mouse.enabled = true
+            canvas._mouse[mapped].connect(function(event) {
+                listener(createEvent(
+                    type,
+                    chart,
+                    event && event.x,
+                    event && event.y,
+                    event,
+                    canvas))
+            })
+        },
+
+        removeEventListener: function(chart, type, listener) {
+            //nothing to do for now
+        }
+    })
+
+    Chart.helpers.merge(Chart.helpers, {
+        color: function(c) {
+            // Implement pattern & gradient if needed
+            return color(c)
+        },
+        getHoverColor: function(c) {
+            // Implement pattern & gradient if needed
+            return Chart.helpers.color(c).saturate(0.5).darken(0.1).rgbString()
```

Comment:
This should probably use `function hoverColor(normalColor)` from ThemePalette.qml

Reply:
Done